### PR TITLE
BaSP-TG-112: Admin form update

### PR DESCRIPTION
**Pull request by GonzaloAsencio**
Replace the actual logic for react hook form logic
Add joi validations

### CODE REVIEW

**Review by MaicolFreire001 — APPROVED** [review]
SGTM🚀

File: src/Components/Admins/editAdminForm.jsx [hunk 1]
```diff
@@ -36,27 +40,31 @@ function Form() {
...
+
   useEffect(async () => {
     if (adminId) {
       document.getElementById('fromHeader').innerHTML = 'EDIT ADMIN';
       if (adminData === undefined) {
         fetch(`${process.env.REACT_APP_API_URL}/admins/${adminId}`)
           .then((response) => response.json())
           .then((response) => {
-            setInputValue({
+            MOCK_DATA = {
```
Comment (JuliMonte5) [inline src/Components/Admins/editAdminForm.jsx:61]: you should do this in a useEffect with a hook that checks for the data to be updated, right now if you are in the edit page and you refresh it (which is the case covered by this fetch) the data in mocked data is not set (the fields get cleared)

File: src/Components/Admins/validations.jsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import Joi from 'joi';
```
Comment (ignacioaristo) [inline src/Components/Admins/validations.jsx:1]: custom all the validation messages.

File: src/Components/Admins/editAdminForm.jsx [hunk 3]
```diff
@@ -36,27 +40,31 @@ function Form() {
     setShowMessageModal(false);
   };
 
+  let MOCK_DATA = {
```
Comment (ignacioaristo) [inline src/Components/Admins/editAdminForm.jsx:43]: Change this variable name, this is not mocked, this is real data

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/Components/Admins/editAdminForm.jsx (hunk 3 above)
Comment (Ignacio-Sormani) [reply src/Components/Admins/editAdminForm.jsx:43]: Also it should be in camelCase. And should probably be a const and not a let

Comment (GonzaloAsencio) [reply src/Components/Admins/editAdminForm.jsx:43]: i forgot to change it before doing the push.Done

File: src/Components/Admins/editAdminForm.jsx (hunk 1 above)
Comment (GonzaloAsencio) [reply src/Components/Admins/editAdminForm.jsx]: done

File: src/Components/Admins/validations.jsx (hunk 2 above)
Comment (GonzaloAsencio) [reply src/Components/Admins/validations.jsx:1]: done

**Review by JuliMonte5 — APPROVED** [review]

**Review by ignacioaristo — APPROVED** [review]

**Review by sofihoj — APPROVED** [review]

**Pull request merged**